run/src/types.ts

```diff
@@ -27,24 +27,24 @@ export type Driver<Si, So> = {
 export type DisposeFunction = () => void;
 
 export type Drivers = {
-  [name: string]: Driver<FantasyObservable | undefined | void | null, any>;
+  [name: string]: Driver<Stream<any> | undefined, any>;
 };
 
 export type Main = (...args: Array<any>) => any;
 
 export type Sources<D extends Drivers> = {[k in keyof D]: ReturnType<D[k]>};
 
 export type Sinks<M extends Main> = {
-  [k in (string & keyof ReturnType<M>)]: ReturnType<M>[k] & FantasyObservable
+  [k in string & keyof ReturnType<M>]: ReturnType<M>[k]
 };
 
 export type MatchingMain<D extends Drivers, M extends Main> = Main & {
-  (so?: Partial<Sources<D>>): Sinks<M>;
+  (so: Sources<D>): Sinks<M>;
 };
```

Comment:
This change is the one that worries me the most, why was it needed? What compile error happens in dom if we revert it?
It's important that we can support `main` functions that either don't use sources or use some of the sources, but this code change means that `main` will have to specify sources always.

Reply:
I have to try again (in case this was not needed in the end) but AFAICR it threw an error if you specified all sources
Maybe a union of both solves this